### ISSUE TITLE
Fix option name compared to API clients & IS:`appId` -> `applicationID`

### ISSUE THREAD

**Issue by redox** [algolia/search-insights.js#265]
Should be easy to keep the backward compat.

**Comment by eunjae-lee** [comment]
@redox When you use search-insights, did you get the type suggestion? If yes, do you think it is not enough and we'd better update the function signature?

<img width="713" alt="image" src="https://user-images.githubusercontent.com/499898/118667279-296f5f00-b7f4-11eb-8cd2-a6fa547bfd10.png">

**Comment by redox** [comment]
> @redox When you use search-insights, did you get the type suggestion? If yes, do you think it is not enough and we'd better update the function signature?

It's more a question of consistency across our library I think, we're using applicationId everywhere we should do here as well. Also, not everyone has autocompletion so I wouldn't rely on this.

**Comment by eunjae-lee** [comment]
> > @redox When you use search-insights, did you get the type suggestion? If yes, do you think it is not enough and we'd better update the function signature?
> 
> It's more a question of consistency across our library I think, we're using applicationId everywhere we should do here as well. Also, not everyone has autocompletion so I wouldn't rely on this.

It'd be more like

```js
aa("init", {
  applicationID: '',
  apiKey: '',
});
```

(The first argument is "init".)

Supporting this alias is not a big deal at all and it's easy to do so. I just want to confirm with you that there are many occurrences and references to `appId` in [the docs](https://www.algolia.com/doc/?q=appid).

**Comment by redox** [comment]
> I just want to confirm with you that there are many occurrences and references to `appId` in [the docs](https://www.algolia.com/doc/?q=appid).

Right, I think we should ultimately unify everything and rename them all; but the most important was the insights one as it's used in combination to our API client and instantsearch JS libraries were we always use "applicationid".

**Comment by Haroenv** [comment]
In InstantSearch we actually used appId back when it was an option: https://github.com/algolia/react-instantsearch/blob/af4b642476f3b3d0b98e0f1afa3d7346fc450893/stories/InstantSearch.stories.js#L11 and https://github.com/algolia/instantsearch.js/blob/v2/src/lib/InstantSearch.js#L40

**Comment by redox** [comment]
> In InstantSearch we actually used appId back when it was an option: https://github.com/algolia/react-instantsearch/blob/af4b642476f3b3d0b98e0f1afa3d7346fc450893/stories/InstantSearch.stories.js#L11 and https://github.com/algolia/instantsearch.js/blob/v2/src/lib/InstantSearch.js#L40
> 
> We have since moved to the option searchClient alone, which indeed has `algoliasearch('appId', 'apiKey')` without named argument

Oh interesting, this is surprising. I still think we need to align it with our docs mentioning everywhere "YourApplicationID" and with our API clients using "application_id/applicationID".

**Comment by eunjae-lee** [comment]
@redox Before you opened this issue, did you have experience where you were using `applicationId` as a parameter somewhere else and come to this repo to see it's `appId` here?

I'm searching for [the doc](https://www.algolia.com/doc/?q=applicationid) and I see the expression "application ID" in plain texts, but not in the code snippets. So I'm not so sure if changing this signature will align with other libraries we provide. Most of the api clients just accept two strings as parameters, although the description says "applicationID" in many cases.

* https://github.com/algolia/algoliasearch-client-javascript: two strings as parameters
* https://github.com/algolia/algoliasearch-client-ruby: two strings
* https://github.com/algolia/algoliasearch-client-scala: two strings
* https://github.com/algolia/algoliasearch-client-csharp: two strings
* https://github.com/algolia/algoliasearch-client-java-2: two strings
* https://github.com/algolia/algoliasearch-client-kotlin: a bit of both (appId and applicationId)
* https://github.com/algolia/algoliasearch-client-python: two strings
* https://github.com/algolia/algoliasearch-client-php: two strings
* https://github.com/algolia/algoliasearch-client-go: two strings
* https://github.com/algolia/algoliasearch-client-swift: `appID`
* https://github.com/algolia/algoliasearch-client-android: two strings

What do you think?

**Comment by sarahdayan** [comment]
@benhinchley In think with the upcoming v3 [this is handled](https://github.com/algolia/search-insights.js/blob/1439d5fec9a3694dff67972c642ee60d55ea9d02/src/insights.ts), so we should be able to close this soon.
